### PR TITLE
🐛 Source S3: fix datetime format string in FileStream

### LINKED ISSUE #22919: S3 source throwing JsonSchemaValidator errors
## Environment
- **Airbyte version**: 0.40.32
- **OS Version / Instance**: macOS
- **Deployment**: Kubernetes
- **Source Connector and version**: S3  0.1.32
- **Destination Connector and version**: Redshift  0.3.56
- **Step where error happened**: Sync job

## Current Behavior
We have successfully been running syncs from S3(avro files) to redshift until a few days ago. We recently upgraded from airbyte 0.39.37 --> 0.40.32 and since then the connection is throwing errors related to JsonSchemaValidator. We have tried using another target but the same error appears. 

For every record the file has a JsonSchemaValidator error is thrown. 

## Expected Behavior
Connection to run without any JsonSchemaValidator errors

## Logs

> `023-02-13 10:25:05 INFO i.a.w.p.KubePodProcess(<init>):592 - Using null stdin output stream...
> 2023-02-13 10:25:05 INFO i.a.w.i.VersionedAirbyteStreamFactory(create):107 - Reading messages from protocol version 0.2.0, messages will be upgraded to protocol version 1.0.0
> 2023-02-13 10:25:05 INFO i.a.w.g.DefaultReplicationWorker(replicate):247 - Waiting for source and destination threads to complete.
> 2023-02-13 10:25:05 INFO i.a.w.g.DefaultReplicationWorker(lambda$readFromDstRunnable$4):275 - Destination output thread started.
> 2023-02-13 10:25:05 INFO i.a.w.g.DefaultReplicationWorker(lambda$readFromSrcAndWriteToDstRunnable$7):337 - Replication thread started.
> 2023-02-13 10:25:12 INFO i.a.v.j.JsonSchemaValidator(test):130 - JSON schema validation failed. 
> errors: $: does not match the regex pattern ^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(Z|[+\-]\d{1,2}:\d{2})( BC)?$
> 2023-02-13 10:25:14 INFO i.a.v.j.JsonSchemaValidator(test):130 - JSON schema validation failed. 
> errors: $: does not match the regex pattern ^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(Z|[+\-]\d{1,2}:\d{2})( BC)?$
> 2023-02-13 10:25:14 INFO i.a.v.j.JsonSchemaValidator(test):130 - JSON schema validation failed. 
> errors: $: does not match the regex pattern ^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(Z|[+\-]\d{1,2}:\d{2})( BC)?$
> 2023-02-13 10:25:14 INFO i.a.v.j.JsonSchemaValidator(test):130 - JSON schema validation failed. `

## Steps to Reproduce
1. Create an S3 source and any kind of destination
2. Run the connection with an avro file and look for JsonSchemaValidator in the logs

## Are you willing to submit a PR?
No

### PR DESCRIPTION
**Pull request by olivermeyer**
## What
Fixes https://github.com/airbytehq/airbyte/issues/22919

The `FileStream` has a `datetime_format_string` attribute which is used to format the cursor field ([here](https://github.com/airbytehq/airbyte/blob/2ce3c17048f6ce66e2c298457e1c0828ce6a4078/airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py#L383)). At some point during the sync, Airbyte seems to validate the format of this field, resulting in errors such as this:
```
errors: $: does not match the regex pattern ^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(Z|[+\-]\d{1,2}:\d{2})( BC)?$
2023-02-17 10:16:14 INFO i.a.v.j.JsonSchemaValidator(test):130 - JSON schema validation failed. 
```
The `FileStream` returns a datetime string formatted as `1970-01-01T00:00+0000`, whereas the validation expects a string formatted as `1970-01-01T00:00Z`.

## How
Fix the datetime format string.

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

**Comment by olivermeyer** [comment]
Attaching the output of `pytest unit_tests`. The integration tests are failing because of Minio timeouts, perhaps they need to run somewhere else.
[s3_unit_tests.txt](https://github.com/airbytehq/airbyte/files/10767120/s3_unit_tests.txt)

**Comment by olivermeyer** [comment]
@natalyjazzviolin could you give this another run? I updated the integration tests.

**Comment by olivermeyer** [comment]
Hi @natalyjazzviolin, pinging you again as we're still seeing issues because of this. Do let me know if there's anything more I can do to get this merged.

**Comment by olivermeyer** [comment]
Hi @natalyjazzviolin, pinging you again as we're still seeing issues because of this. Do let me know if there's anything more I can do to get this merged.

**Comment by olivermeyer** [comment]
The latest commit should fix the issue of comparing timezone-aware datetime objects with unaware objects. Attaching unit test logs for this commit.
[unittests.log](https://github.com/airbytehq/airbyte/files/10919611/unittests.log)

**Comment by evantahler** [comment]
@sh4sh 👍 from me, assuming the tests pass.
@edgao When double-checking the format of `date-time`, I found that https://docs.airbyte.com/understanding-airbyte/supported-data-types seems to only include the V2 types now.  Can you confirm that this change is the right move for the current types?

**Review by edgao — APPROVED** [review]
is this change switching the TZ from `+0000` to `Z`? That seems right to me. Some destinations (e.g. snowflake) assume that the tz offset is either Z or +/- `HH:MM` (i.e. with a colon), e.g. https://github.com/airbytehq/oncall/issues/1476.

File: docs/integrations/sources/s3.md [hunk 1]
```diff
@@ -209,6 +209,7 @@ The Jsonl parser uses pyarrow hence,only the line-delimited JSON format is suppo
 
 | Version | Date       | Pull Request                                                                                                    | Subject                                                                                    |
 |:--------|:-----------|:----------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------|
+| 1.0.3   | 2023-03-06 | [23195](https://github.com/airbytehq/airbyte/pull/23195)                                                        | Fix datetime format string                                                                 |
```
Comment (evantahler) [inline docs/integrations/sources/s3.md:212]: @olivermeyer this PR is one file change away from being ready to merge - You also need to bump the version in the [Dockerfile](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-s3/Dockerfile).

**Review by evantahler — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/s3.md (hunk 1 above)
Comment (olivermeyer) [reply docs/integrations/sources/s3.md:212]: Done :)

**Comment by evantahler** [comment]
Oh no @olivermeyer - we've got merge conflicts again.  `source-s3` just had v2 released.  I'll settle the merge conflicts and re-run the tests.  Looks like you are creating v2.0.1 now

**Review by evantahler — APPROVED** [review]
Tests passing, let's go!

**Pull request merged**